src/forces/Gravity.re

```diff
@@ -2,16 +2,24 @@ let g = 6.67428 *. Math.pow(~base=10, ~exp=-11);
 let force = (~attractorMass, ~moverMass, ~r) =>
   g *. attractorMass *. moverMass /. Math.sqf(r);
 
-let forceV = (~attractorMass, ~moverMass, ~attractor, ~mover) => {
+let forceV =
+    (~attractorMass, ~moverMass, ~attractor, ~mover, ~threshold=?, ()) => {
   // Derive the vector pointing from attractor to mover.
   let v = Vector.subf(~v1=attractor, ~v2=mover);
 
   // Derive the magnitude of the above vector.
   let mag = Vector.magf(v);
 
+  // Constrain the gravitational force if threshold constraints were supplied.
+  let distance =
+    switch (threshold) {
+    | Some(th) => Math.constrain(~low=fst(th), ~high=snd(th), mag)
+    | None => mag
+    };
```

Comment:
My attempt to constrain the gravitational force. Basically, if the `distance` between `mover` and `attractor` exceeds the range, we use the clamped distance to calculate the force to apply. I'm missing something here I think, because this will constrain the force but not necessarily the position of the `mover`.